lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3338,17 +3340,26 @@ void PostVisitBlock(ParseNode *pnode, ByteCodeGenerator *byteCodeGenerator)
         return;
     }
 
+    Scope *scope = pnode->sxBlock.scope;
     if (pnode->sxBlock.GetCallsEval() || pnode->sxBlock.GetChildCallsEval() || (byteCodeGenerator->GetFlags() & (fscrEval | fscrImplicitThis | fscrImplicitParents)))
     {
-        Scope *scope = pnode->sxBlock.scope;
         bool scopeIsEmpty = scope->IsEmpty();
         scope->SetIsObject();
         scope->SetCapturesAll(true);
         scope->SetMustInstantiate(!scopeIsEmpty);
     }
 
-    byteCodeGenerator->PopScope();
-    byteCodeGenerator->PopBlock();
+    if (scope->GetScopeType() != ScopeType_Parameter)
+    {
+        byteCodeGenerator->PopScope();
+        byteCodeGenerator->PopBlock();
+    }
+    else
+    {
+        // If current block is a param scope then it was not pushed in during PreVisitBlock. So let's not do a pop here.
```

Comment:
What now?  The spec calls for the body scope to be nested in the parameter scope.  Why do we want to make them siblings?  Won't that require special casing look-ups everywhere?